src/imitation/algorithms/sqil.py

```diff
@@ -0,0 +1,314 @@
+            device: Device (cpu, cuda, ...) on which the code should be run.
+                Setting it to auto, the code will be run on the GPU if possible.
+            _init_setup_model: Whether or not to build the network
+                at the creation of the instance.
+
+
+        """
+        self.venv = venv
+
+        super().__init__(demonstrations=demonstrations, custom_logger=custom_logger)
+
+        self.dqn = dqn.DQN(
+            policy=policy,
+            env=venv,
+            learning_rate=learning_rate,
+            buffer_size=buffer_size,
+            learning_starts=learning_starts,
+            batch_size=batch_size,
+            tau=tau,
+            gamma=gamma,
+            train_freq=train_freq,
+            gradient_steps=gradient_steps,
+            replay_buffer_class=replay_buffer_class,
+            replay_buffer_kwargs=replay_buffer_kwargs,
+            optimize_memory_usage=optimize_memory_usage,
+            target_update_interval=target_update_interval,
+            exploration_fraction=exploration_fraction,
+            exploration_initial_eps=exploration_initial_eps,
+            exploration_final_eps=exploration_final_eps,
+            max_grad_norm=max_grad_norm,
+            tensorboard_log=tensorboard_log,
+            policy_kwargs=policy_kwargs,
+            verbose=verbose,
+            seed=seed,
+            device=device,
+            _init_setup_model=_init_setup_model,
+        )
+
+    def set_demonstrations(self, demonstrations: algo_base.AnyTransitions) -> None:
+        # If demonstrations is a list of trajectories,
+        # flatten it into a list of transitions
+        if isinstance(demonstrations, Iterable):
+            (
+                item,
+                demonstrations,
+            ) = util.get_first_iter_element(  # type: ignore[assignment]
+                demonstrations,  # type: ignore[assignment]
+            )
+            if isinstance(item, types.Trajectory):
+                demonstrations = rollout.flatten_trajectories(
+                    demonstrations,  # type: ignore[arg-type]
+                )
+
+        n_samples = len(demonstrations)  # type: ignore[arg-type]
+        self.expert_buffer = buffers.ReplayBuffer(
+            n_samples,
+            self.venv.observation_space,
+            self.venv.action_space,
+            handle_timeout_termination=False,
+        )
+
+        for transition in demonstrations:
+            self.expert_buffer.add(
+                obs=np.array(transition["obs"]),  # type: ignore[index]
+                next_obs=np.array(transition["next_obs"]),  # type: ignore[index]
+                action=np.array(transition["acts"]),  # type: ignore[index]
+                done=np.array(transition["dones"]),  # type: ignore[index]
+                reward=np.array(1),
+                infos=[{}],
+            )
+
+    def train(self, *, total_timesteps: int):
+        self.learn_dqn(total_timesteps=total_timesteps)
```

Comment:
Outdated now, but my idea here was keeping some level of consistency in the API. It's a bit awkward since in SB3, `train` performs one step and `learn` does multiple steps; and in imitation, `train` performs multiple steps. But imo the important thing is the consistency of the API